app/root.tsx

```diff
@@ -91,6 +101,14 @@ export default function App() {
             </analyticsFetcher.Form>
           </div>
         )}
+        {process.env.NODE_ENV === "development" || !gaTrackingId ? null : (
+          <>
+            <script
+              async
+              src={`https://www.googletagmanager.com/ns.html?id=${gaTrackingId}`}
+            />
+          </>
+        )}
```

Comment:
We don't need the empty <> because we only have one object inside. They would be needed to wrap more than one object AFAIK :)